src/main/java/utils/HttpRequest.java

```diff
@@ -0,0 +1,152 @@
+import org.apache.http.entity.StringEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+
+@Accessors(chain = true)
+public class HttpRequest {
+
+    private HttpRequestBase request;
+    private HttpExecutor executor = HttpExecutor.getInstance();
+
+    public enum HttpMethod {
+        GET, POST, DELETE, PUT
+    }
+
+    @Getter
+    @Setter
+    protected String jenkinsBaseUrl;
+
+    @Getter
+    @Setter
+    protected String endpoint;
+    @Getter
+    protected Map<String, String> params;
+
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpRequest.class);
+
+    public HttpRequest(HttpMethod method) {
+        params = new HashMap<>();
+
+        switch (method) {
+        case POST:
+            request = new HttpPost();
+            break;
+        case GET:
+            request = new HttpGet();
+            break;
+        case DELETE:
+            request = new HttpDelete();
+            break;
+        case PUT:
+            request = new HttpPut();
+            break;
+        }
+    }
+
+    /*
+     * Function that clean parameters field only.
+     */
+    public void cleanParams() {
+        params.clear();
+    }
+
+    /*
+     * Function that resets the HTTP Request object so it can be reused.
+     *
+     */
+    public void resetHttpRequestObject() {
+        this.cleanParams();
+        request.reset();
+    }
+
+    /*
+     * Function for adding headers to the http request.
+     *
+     * @param key , the key of the header
+     * @param value, the value of the header
+     *
+     * @return HTTPRequest
+     */
+    public HttpRequest addHeader(String key, String value) {
+        request.addHeader(key, value);
+        return this;
+    }
+
+    /*
+     * Function for adding parameters to the http request.
+     *
+     * @param key , the key of the parameter
+     * @param value, the value of the parameter
+     *
+     * @return HTTPRequest
+     */
+    public HttpRequest addParam(String key, String value) {
+        params.put(key, value);
+        return this;
+    }
+
+    /*
+     * Function that set the body of the http request.
+     *
+     * @param body , the body to be set in the http request.
+     *
+     * @return HTTPRequest
+     */
+    public HttpRequest setBody(String body) {
+        ((HttpEntityEnclosingRequestBase) request).setEntity(new StringEntity(body, "UTF-8"));
+        return this;
+    }
+
+    /*
+     * Function that set the body of the http request.
+     *
+     * @param body , the file with body content to be set in the http request.
+     */
+    public void setBody(File file) {
+        String fileContent = "";
+        try {
+            fileContent = FileUtils.readFileToString(file, "UTF-8");
+        } catch (IOException e) {
+            LOGGER.error("Failed to read the Request body file:" + file.getPath() + ". Message: " + e.getMessage(), e);
```

Comment:
Should this maybe throw an error? It feels like if this goes wrong, the code should not continue?